src/Database/Commands/IndexQueryBasedCommand.ts

```diff
@@ -0,0 +1,50 @@
+import {RequestMethod} from "../../Http/Request/RequestMethod";
+import {IndexQuery} from "../Indexes/IndexQuery";
+import {QueryOperationOptions} from "../Operations/QueryOperationOptions";
+import {RavenCommand} from "../RavenCommand";
+import {ServerNode} from "../../Http/ServerNode";
+import {TypeUtil} from "../../Utility/TypeUtil";
+import {InvalidOperationException} from "../DatabaseExceptions";
+import {StringUtil} from "../../Utility/StringUtil";
+
+export abstract class IndexQueryBasedCommand extends RavenCommand {
+  protected indexName?: string;
+  protected query?: IndexQuery;
+  protected options?: QueryOperationOptions;
+
+  constructor(method: RequestMethod, indexName: string, query: IndexQuery, options?: QueryOperationOptions) {
+    super('', method);
+    this.indexName = indexName;
+    this.query = query;
+    this.options = options || new QueryOperationOptions();
+  }
+
+  public createRequest(serverNode: ServerNode): void {
+    const query: IndexQuery = this.query;
+    const options: QueryOperationOptions = this.options;
+
+    if (TypeUtil.isNone(this.indexName)) {
+      throw new InvalidOperationException('Empty index_name is not valid');
+    }
+
+    if (!(query instanceof IndexQuery)) {
+      throw new InvalidOperationException('Query must be instance of IndexQuery class');
+    }
+
+    if (!(options instanceof QueryOperationOptions)) {
+      throw new InvalidOperationException('Options must be instance of QueryOperationOptions class');
+    }
+
+    this.params = {
+      pageSize: query.pageSize,
+      allowStale: options.allowStale,
+      details: options.retrieveDetails
+    };
+
+    this.endPoint += StringUtil.format('/queries/{0}', this.indexName);
+    query.query && this.addParams('query', query.query);
```

Comment:
You can see in the session_query.py 
the non stale option is in the __call__ function 

```
def __call__(self, object_type=None, index_name=None, using_default_operator=None,
                 wait_for_non_stale_results=False, includes=None, with_statistics=False, nested_object_types=None):
```
wait_for_non_stale_results (see test_query.py)
Other params can be found in the __call__ or in the function below